### ISSUE TITLE
[Question] Error with the contruction of the new_label？

### ISSUE THREAD

**Issue by ygfrancois** [haotian-liu/LLaVA#688]
### Question

https://github.com/haotian-liu/LLaVA/blob/main/llava/model/llava_arch.py#L151

`            while image_token_indices.numel() > 0:
                cur_image_features = image_features[cur_image_idx]
                image_token_start = image_token_indices[0]
                if getattr(self.config, 'tune_mm_mlp_adapter', False) and getattr(self.config, 'mm_use_im_start_end', False):
                    cur_new_input_embeds.append(self.get_model().embed_tokens(cur_input_ids[:image_token_start-1]).detach())
                    cur_new_input_embeds.append(self.get_model().embed_tokens(cur_input_ids[image_token_start-1:image_token_start]))  # 对special token做单独的embed
                    cur_new_input_embeds.append(cur_image_features)
                    cur_new_input_embeds.append(self.get_model().embed_tokens(cur_input_ids[image_token_start+1:image_token_start+2]))
                    if labels is not None:
                        cur_new_labels.append(cur_labels[:image_token_start])
                        cur_new_labels.append(torch.full((cur_image_features.shape[0],), IGNORE_INDEX, device=labels.device, dtype=labels.dtype))
                        cur_new_labels.append(cur_labels[image_token_start:image_token_start+1])
                        cur_labels = cur_labels[image_token_start+2:]`
                        

why   cur_new_labels.append(cur_labels[image_token_start:image_token_start+1])   ？

cur_labels[image_token_start:image_token_start+1] means the IMAGE_TOKEN_INDEX=-200，but I think it should be im_end token ?

**Comment by dreamerlin** [comment]
Same question with u

**Comment by haotian-liu** [comment]
Uh I think you are right. But it would not cause any problem for the current training cases, because they are all masked and the *labels* should be all `IGNORE_INDEX`.
